react-accordion-component/src/accordion.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+class AccordionSegment extends React.Component {
+  render() {
+    return (this.props.content === '')
+      ? (
+        <div id={this.props.identifier} className="accordion-segment">
+          <h2 onClick={this.props.handleClick} className="segment-title">{this.props.title}</h2>
+        </div>
+        )
+      : (
+        <div id={this.props.identifier} className="accordion-segment">
+          <h2 onClick={this.props.handleClick} className="segment-title">{this.props.title}</h2>
+          <p className="segment-content">{this.props.content}</p>
+        </div>
+        );
+  }
+}
+
+export default class Accordion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.clickHandler = this.clickHandler.bind(this);
+    this.state = { active: '' };
+  }
+
+  clickHandler(event) {
+    const id = event.target.closest('div').id;
+    if (id === this.state.active) {
+      this.setState({ active: '' });
+    } else {
+      this.setState({ active: id });
+    }
+  }
+
+  render() {
+    let index = -1;
+    const topicsList = this.props.topics.map(i => {
+      index++;
```

Comment:
Instead of using your own index you can use the one passed as a parameter to the callback of `map`:

```JAVASCRIPT
 const topicsList = this.props.topics.map((item, index) => {
 ```